### ISSUE TITLE
Error updating resource Key=IPAMBlock

### ISSUE THREAD

**Issue by cloud-66** [projectcalico/calico#4726]
pods stuck in CreatingContainer status. Аfter a long research, I found the errors in containerd debug log
There is problem than number of pods per node more than 120-130 pods  (maxPods: 180)
```
ERROR][4138697] customresource.go 136: Error updating resource Key=IPAMBlock(10-244-101-0-24) Name="10-244-101-0-24"  
Resource="IPAMBlocks" Value=&v3.IPAMBlock{TypeMeta:v1.TypeMeta{Kind:"IPAMBlock", APIVersion:"crd.projectcalico.org/v1"},   
ObjectMeta:v1.ObjectMeta{Name:"10-244-101-0-24", GenerateName:"", Namespace:"", SelfLink:"", UID:"",  
ResourceVersion:"9994410", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}},     DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil),    
Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"",  
ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v3.IPAMBlockSpec{CIDR:"10.244.101.0/24", Affinity:(*string)(0xc0003cf850),   
StrictAffinity:false, Allocations:[]*int{(*int)(0xc000601e40)
```

after that
```
Jul  7 11:11:57 qrunner-p01 containerd[3818731]: time="2021-07-07T11:11:57.203527549Z" level=info msg="starting signal   
loop" namespace=k8s.io path=/run/containerd/io.containerd.runtime.v2.task/k8s.io  
/b1246f8bfc4143cc68be152a6643b699d1a11be574642c1b3d6db4783171f981 pid=4138824  
Jul  7 11:11:57 qrunner-p01 kubelet[4127983]: I0707 11:11:57.210347 4127983 kubelet.go:1868] SyncLoop (UPDATE, "api"): "load-  
autotest-ftfabui-dev-mdpauth-79669c6d59-pdjtv_load-autotest-ftfabui(16e6caad-31f3-44cd-96e2-bbab3b1f4125)"  
Jul  7 11:11:57 qrunner-p01 systemd[1]: Started libcontainer container   
b1246f8bfc4143cc68be152a6643b699d1a11be574642c1b3d6db4783171f981.
```

```
Jul  7 11:12:01 qrunner-p01 containerd[3818731]: time="2021-07-07T11:12:01.453737611Z" level=error msg="get state for b1246f8bfc4143cc68be152a6643b699d1a11be574642c1b3d6db4783171f981" error="context deadline exceeded: unknown"
```

## Expected Behavior
correctly started pods

## Current Behavior
pods stuck in CreatingContainer status

## Steps to Reproduce (for bugs)
1.  run more than 130-140 pods per one node

## Your Environment

* Calico version
v3.17.4
kubernetes v1.19.8 installed by kubespray 2.16.0
containerd v1.4.4 as runtime
kubespray options
typha_enabled: true
calico_ipip_mode: 'CrossSubnet'
kube_service_addresses: 10.233.0.0/16
kube_pods_subnet: 10.244.0.0/16
kube_network_node_prefix: 24

* Operating System and version:
Ubuntu 20.04.2

**Comment by song-jiang** [comment]
@cloud-66 You saw this issue happening only when number of pods running on the node is more than 120? There is no problem for the previous 1 - 119 pods to run?

**Comment by cloud-66** [comment]
This situation was bug in runc 1.0-rc93 version which is included to containerd 1.4.4 . With containerd 1.4.6 is much better.
